JDBC/OAuthSampleApp/src/main/java/OAuthSampleApp.java

```diff
@@ -10,91 +10,329 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
-
 import java.sql.Connection;
 import java.sql.DriverManager;
 import java.sql.ResultSet;
 import java.sql.SQLException;
 import java.sql.Statement;
 import java.sql.SQLTransientConnectionException;
 import java.sql.SQLInvalidAuthorizationSpecException;
+import java.util.Map;
+import java.util.HashMap;
 import java.util.Properties;
-import java.util.concurrent.Callable;
-
-import picocli.CommandLine;
-import picocli.CommandLine.Command;
-import picocli.CommandLine.Option;
-import picocli.CommandLine.Parameters;
-
-@Command(name = "OAuthSampleApp", mixinStandardHelpOptions = true, version = "OAuth sample app 1.0", description = "Connects to a Vertica database using OAuth")
-public class OAuthSampleApp implements Callable<Integer> {
-
-    @Parameters(index = "0", description = "Host name")
-    private String host = "";
-
-    @Parameters(index = "1", description = "Database name")
-    private String dbName = "";
-
-    @Option(names = { "-p", "--port" }, description = "Port")
-    private String port = "5433";
-
-    @Option(names = { "-a", "--access-token" }, description = "Access token")
-    private String accessToken = "";
-
-    @Option(names = { "-r", "--refresh-token" }, description = "Refresh token")
-    private String refreshToken = "";
-
-    @Option(names = { "-s", "--client-secret" }, description = "Client Secret")
-    private String clientSecret = "";
-
-    private static Connection connectToDB(String host, String port, String dbName, String accessToken,
-            String refreshToken, String clientSecret) throws SQLException {
-        Properties jdbcOptions = new Properties();
-        jdbcOptions.put("oauthaccesstoken", accessToken);
-        jdbcOptions.put("oauthrefreshtoken", refreshToken);
-        jdbcOptions.put("oauthclientsecret", clientSecret);
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.FileInputStream;
+import java.io.BufferedInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.net.URLEncoder;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.vertica.jdbc.VerticaConnection;
 
-        return DriverManager.getConnection(
-                "jdbc:vertica://" + host + ":" + port + "/" + dbName, jdbcOptions);
+public class OAuthSampleApp
+{
+	private static Properties prop;
+	private static String OAUTH_ACCESS_TOKEN_VAR_STRING = "OAUTH_SAMPLE_ACCESS_TOKEN";
+	private static String OAUTH_REFRESH_TOKEN_VAR_STRING = "OAUTH_SAMPLE_REFRESH_TOKEN";
+	private static Connection vConnection;
+	// Get jdbc connection string from provided configuration
+	private static String getConnectionString(String args) 
```

Comment:
You can simplify this a lot by removing getConnectionString and getParam. Just put a sample connection string in the config properties like jdbc:vertica://localhost:5433/VMart or whatever you use.